CompoundAssignmentExpr Parser bug for |=

```
beginning null denotation identifier handling
current peek token when starting path pratt parse: '%='
current token (just about to return path to null denotation): '%='
finished null denotation identifier path parsing - next is branching 
test.rs:21:9: error: found unexpected token ‘=’ in null denotation
   21 |     e %= 5;
      |         ^
null denotation is null; returning null for parse_expr
left denotation is null; returning null for parse_expr
test.rs:21:5: error: failed to parse expression for expression without block (pratt-parsed expression is null)
   21 |     e %= 5;
      |     ^
test.rs:21:5: error: failed to parse statement or expression without block in block expression
test.rs:21:11: error: unrecognised token ‘integer literal’ for start of item
   21 |     e %= 5;
      |           ^

```

The parser only seems to fail with:

```
    e %= 5;
    g |= 7;
    h ^= 8;
```

%= and ^= are fixed in #175 

|= shows:

```
```
beginning null denotation identifier handling
current peek token when starting path pratt parse: '|='
current token (just about to return path to null denotation): '|='
finished null denotation identifier path parsing - next is branching 
test.rs:24:7: error: expecting ‘}’ but ‘|=’ found
   24 |     g |= 7;
      |       ^
test.rs:24:5: error: error may be from having an expression (as opposed to statement) in the body of the function but not last
   24 |     g |= 7;
      |     ^

```
```

--- a/gcc/rust/parse/rust-parse-impl.h
+++ b/gcc/rust/parse/rust-parse-impl.h
@@ -386,6 +386,8 @@ Parser<ManagedTokenSource>::left_binding_power (const_TokenPtr token)
       return LBP_MOD_ASSIG;
     case AMP_EQ:
       return LBP_AMP_ASSIG;
+    case PIPE_EQ:
+      return LBP_PIPE_ASSIG;
     case CARET_EQ:
       return LBP_CARET_ASSIG;
     case LEFT_SHIFT_EQ:

--- a/gcc/testsuite/rust.test/compilable/compound_assignment_expr1.rs
+++ b/gcc/testsuite/rust.test/compilable/compound_assignment_expr1.rs
@@ -16,8 +16,7 @@ fn main() {
     d /= 4;
     e %= 5;
     f &= 6;
-    // https://github.com/Rust-GCC/gccrs/issues/173
-    // g |= 7;
+    g |= 7;
     h ^= 8;
     i <<= 9;
     j >>= 10;

